lib/utils.js

```diff
@@ -0,0 +1,46 @@
+var Q = require('q');
+var es = require('.././config/es');
+
+var log = console.log;
+
+exports = module.exports = {
+  waitForElastic: function (waitCnt) {
+    var waitCntDef = 50;
+    waitCnt = waitCnt || 50;  // 40 requests by default
+
+    var deferred = Q.defer();
+
+    (function pinger(done) {
+      if(waitCnt != waitCntDef) log('requests left', waitCnt);
+
+      if(waitCnt-- == 0) {
+        deferred.reject(new Error('Max requests exceeded, elastic is dead.'));
```

Comment:
What do you think about this code? Seems it's better to use elasticsearch echo timeout that our setTimeout.

```javascript
waitForElastic: function(waitCounter) {
	waitCounter = waitCounter || 50;
	var REQUEST_TIMEOUT = 1000;

	var deferred = Q.defer();

	var pingUntilSuccessOrZeroCounter = function() {
		waitCounter--;
				
		if (waitCounter === 0) {
			 deferred.failure("Max requests exceeded");
		} else {
			elastic
				.ping({
					requestTimeout: REQUEST_TIMEOUT
				})
				.then(deferred.resolve, pingUntilSuccessOrZeroCounter);
		}
	};

	pingUntilSuccessOrZeroCounter();

	return deferred.promise;
}
```